include/picongpu/plugins/Emittance.hpp

```diff
@@ -0,0 +1,1050 @@
+             */
+            pmacc::GridController< simDim >& gc = pmacc::Environment< simDim >::get( ).GridController( );
+            pmacc::math::Size_t< simDim > gpuDim = gc.getGpuNodes( );
+            pmacc::math::Int< simDim > gpuPos = gc.getPosition( );
+
+            /* my plane means: the r_element I am calculating should be 1GPU in width */
+            pmacc::math::Size_t< simDim > sizeTransversalPlane( gpuDim );
+            sizeTransversalPlane[ r_element ] = 1;
+
+            // avoid deadlock for following, blocking MPI operations
+            __getTransactionEvent( ).waitForFinished( );
+
+            for( int planePos = 0; planePos <= ( int )gpuDim[ r_element ]; ++planePos )
+            {
+                /* my plane means: the offset for the transversal plane to my r_element
+                 * should be zero
+                 */
+                pmacc::math::Int< simDim > longOffset( pmacc::math::Int< simDim >::create( 0 ) );
+                longOffset[ r_element ] = planePos;
+
+                zone::SphericZone< simDim > zoneTransversalPlane( sizeTransversalPlane, longOffset );
+
+                /* Am I the lowest GPU in my plane? */
+                bool isGroupRoot = false;
+                bool isInGroup = ( gpuPos[ r_element ] == planePos );
+                if( isInGroup )
+                {
+                    pmacc::math::Int< simDim > inPlaneGPU( gpuPos );
+                    inPlaneGPU[ r_element ] = 0;
+                    if( inPlaneGPU == pmacc::math::Int< simDim >::create( 0 ) )
+                        isGroupRoot = true;
+                }
+                algorithm::mpi::Reduce< simDim >* createReduce =
+                    new algorithm::mpi::Reduce< simDim >( zoneTransversalPlane,
+                                                        isGroupRoot );
+                if( isInGroup )
+                {
+                    planeReduce = createReduce;
+                    isPlaneReduceRoot = isGroupRoot;
+                }
+                else
+                    __delete( createReduce );
+            }
+
+            /* Create communicator with ranks of each plane reduce root */
+            {
+                /* Array with root ranks of the planeReduce operations */
+                std::vector< int > planeReduceRootRanks( gc.getGlobalSize( ), -1 );
+                /* Am I one of the planeReduce root ranks? my global rank : -1 */
+                int myRootRank = gc.getGlobalRank( ) * isPlaneReduceRoot
+                    - ( ! isPlaneReduceRoot );
+
+                MPI_Group world_group, new_group;
+                MPI_CHECK(
+                    MPI_Allgather(
+                        &myRootRank,
+                        1,
+                        MPI_INT,
+                        planeReduceRootRanks.data( ),
+                        1,
+                        MPI_INT,
+                        MPI_COMM_WORLD
```

Comment:
Since `gc.getGlobalRank( )` is used to get the rank `MPI_COMM_WORLD` must be replaced by `gc.getCommunicator().getMPIComm()`. 

Why?: The rank calculated in the grid controller `gc` based on an own communicator created for PMacc. MPI allows that the ranks enumeration differs in each communicator world.